images/tools/Dockerfile

```diff
@@ -1,41 +1,76 @@
-# Copyright IBM Corp. All Rights Reserved.
+#
+# Copyright contributors to the Hyperledger Fabric project
 #
 # SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at:
+#
+# 	  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
 
+###############################################################################
+# Build image
+###############################################################################
+
+ARG UBUNTU_VER
+FROM ubuntu:${UBUNTU_VER} as builder
+
+ARG TARGETARCH
+ARG TARGETOS
+ARG FABRIC_VER
 ARG GO_VER
-ARG ALPINE_VER
-FROM golang:${GO_VER}-alpine${ALPINE_VER} as golang
+ARG GO_TAGS
 
-RUN apk add --no-cache \
-	bash \
-	binutils-gold \
-	gcc \
-	git \
-	make \
-	musl-dev;
+RUN apt update
+RUN apt install -y git
+RUN apt install -y gcc
+RUN apt install -y curl
+RUN apt install -y make
+RUN curl -sL https://go.dev/dl/go${GO_VER}.${TARGETOS}-${TARGETARCH}.tar.gz | tar zxf - -C /usr/local
+ENV PATH="/usr/local/go/bin:$PATH"
 
-ADD . $GOPATH/src/github.com/hyperledger/fabric
-WORKDIR $GOPATH/src/github.com/hyperledger/fabric
+ADD . .
 
-FROM golang as tools
-ARG GO_TAGS
+RUN make tools GO_TAGS=${GO_TAGS} FABRIC_VER=${FABRIC_VER}
+
+
+###############################################################################
+# Runtime image
+###############################################################################
+
+ARG UBUNTU_VER
+FROM ubuntu:${UBUNTU_VER}
+
+ARG TARGETOS
+ARG TARGETARCH
 ARG FABRIC_VER
-ENV FABRIC_VER ${FABRIC_VER}
-
-# Disable cgo when building in the container.  This will create a static binary, which can be
-# copied and run in the base alpine container without the libc/musl runtime.
-ENV CGO_ENABLED 0
```

Comment:
Why did you remove static compilation?

Reply:
The static link in docker was only added to work around the SIGSEGVs in alpine.  Now with the build in Docker and with the matching libc runtime FROM ubuntu, the static build is no longer required.

Reply:
Compiling binary statically has another side effect of having all dependencies compiled into binary, ensuring the resulting executable is assembled and equipped with all dependencies it was compiled with, ensuring consistency.

Reply:
@C0rWin are you proposing that we change the `make` target to generate a static executable?  I thought we had settled on the use of Docker to establish consistency across runtimes.  Recall that the above update `ENV CGO_ENABLED 0` was _new behavior_ introduced in the release 2.5 alpha builds for alpine - the behavior in this PR reverts to the same mechanisms we've been using in Fabric since ~2017, with the primary difference that the base Docker image is now derived from ubuntu.  Running with the ubuntu / gnu libc seems to have resolved all of the errors encountered with alpine's musl runtime.

Also: 

- Formal release binaries are static, generated with the golang cross-compiler.  

- Changing the default `make` target to a static build will cause major issues for the pkcs11 runtimes.